app/concepts/repositories/halls.rb

```diff
@@ -0,0 +1,29 @@
+module Repositories
+    class Halls
+        attr_reader :adapter
+
+        def initialize(adapter: Hall)
+            @adapter = adapter
+        end
+
+        def find_all
+            adapter.order(:id).all
+        end
+
+        def find(id)
+            adapter.find(id)
+        end
+
+        def create(params)
+            adapter.create(params)
+        end
+
+        def update(hall, params)
+            hall.update(params)
```

Comment:
Remember to not base on the object itself in the repository(all operations should be done by the adapter). Consider refactor to
```
def update(id, params)
  adapter.find(id).update(params)
end
```
or even better (with reusing methods):
```
def update(id, params)
  find(id).update(params)
end
```

Reply:
I had it this way but then it only return true/false and I needed the hall object in my controller. How could I achieve this differently?

Reply:
You're right, I've forgot about it. Could you try this usage of `update` method?
```
def update(id, params)
  adapter.update(id, params)
end
```

Reply:
So I changed this update method in repository, but then I should my action look like?

```
    def update
        @hall = UseCases::Halls::Update.new.call(id: params[:id], params: hall_params)
        
        if @hall.valid?
            render json: @hall
        else
            render json: @hall.errors, status: :unprocessable_entity
        end
    end
```

Reply:
Exactly. It's more readable and we reduced the number of requests to the database.

Btw. sometimes it's a good idea to prevalidate if the object exists directly in the controller or in the first step in the use case. It's important when there is a bigger business process instead of simple `#update` method usage, but at this point, I believe that this is a good solution 👍